File: Android/SmartMicrogrid.Android/app/src/main/java/com/smartmicrogrid/models/EnergySlot.kt
package com.smartmicrogrid.models

import com.google.gson.annotations.SerializedName

data class EnergySlot(
    @SerializedName("id") val id: String = "",
    @SerializedName("microgridNodeId") val microgridNodeId: String = "",
    @SerializedName("microgridName") val microgridName: String = "",
    @SerializedName("location") val location: String = "",
    @SerializedName("energyAmount") val energyAmount: Double = 0.0,
    @SerializedName("availableAmount") val availableAmount: Double = 0.0,
    @SerializedName("startTime") val startTime: String = "",
    @SerializedName("endTime") val endTime: String = "",
    @SerializedName("pricePerUnit") val pricePerUnit: Double = 0.0,
    @SerializedName("status") val status: String = "Available",
    @SerializedName("createdBy") val createdBy: String = "",
    @SerializedName("createdAt") val createdAt: String = "",
    @SerializedName("updatedAt") val updatedAt: String = ""
)

data class CreateEnergySlotRequest(
    @SerializedName("microgridNodeId") val microgridNodeId: String,
    @SerializedName("energyAmount") val energyAmount: Double,
    @SerializedName("availableAmount") val availableAmount: Double,
    @SerializedName("startTime") val startTime: String,
    @SerializedName("endTime") val endTime: String,
    @SerializedName("pricePerUnit") val pricePerUnit: Double,
    @SerializedName("status") val status: String = "Available"
)

data class EnergyAvailabilitySlot(
    @SerializedName("energySlotId") val energySlotId: String = "",
    @SerializedName("microgridNodeId") val microgridNodeId: String = "",
    @SerializedName("microgridName") val microgridName: String = "",
    @SerializedName("location") val location: String = "",
    @SerializedName("energyAmount") val energyAmount: Double = 0.0,
    @SerializedName("availableAmount") val availableAmount: Double = 0.0,
    @SerializedName("startTime") val startTime: String = "",
    @SerializedName("endTime") val endTime: String = "",
    @SerializedName("pricePerUnit") val pricePerUnit: Double = 0.0,
    @SerializedName("status") val status: String = "Available"
)

File: Backend/SmartMicrogrid.API/Models/M1/EnergySlot.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace SmartMicrogrid.API.Models.M1
{
    public class EnergySlot
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("microgridNodeId")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string MicrogridNodeId { get; set; } = string.Empty;

        [BsonElement("energyAmount")]
        public double EnergyAmount { get; set; }

        [BsonElement("availableAmount")]
        public double AvailableAmount { get; set; }

        [BsonElement("startTime")]
        public DateTime StartTime { get; set; }

        [BsonElement("endTime")]
        public DateTime EndTime { get; set; }

        [BsonElement("pricePerUnit")]
        public decimal PricePerUnit { get; set; }

        [BsonElement("status")]
        public string Status { get; set; } = "Available"; // Available, PartiallyReserved, FullyReserved, Expired, Cancelled

        [BsonElement("createdBy")]
        public string CreatedBy { get; set; } = string.Empty;

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [BsonElement("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
